File: dotnet-desktop-guide/net/winforms/printing/snippets/overview/vb/Form1.vb
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        PrintDialog1.Document = PrintDocument1
        '<show_dialog>
        If PrintDialog1.ShowDialog() = DialogResult.OK Then
            PrintDocument1.Print()
        End If
        '</show_dialog>
    End Sub
    '<specify_the_material_to_be_printed>
    Private Sub PrintDocument1_PrintPage(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintPageEventArgs) Handles PrintDocument1.PrintPage
        e.Graphics.FillRectangle(Brushes.Red, New Rectangle(100, 100, 100, 100))
    End Sub
    '</specify_the_material_to_be_printed>

    '<message_box_indicating_document_has_finished_printing>
    Private Sub PrintDocument1_EndPrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles PrintDocument1.EndPrint
        MessageBox.Show(PrintDocument1.DocumentName + " has finished printing.")
    End Sub
    '</message_box_indicating_document_has_finished_printing>

End Class

File: dotnet-desktop-guide/net/winforms/printing/snippets/overview/csharp/Form1.cs
namespace Sample_Print_Application
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            printDialog1.Document = printDocument1;
            printDocument1.DocumentName = "SamplePrintApp";
            //<show_dialog>
            // display show dialog and if user selects "Ok" document is printed
            if (printDialog1.ShowDialog() == DialogResult.OK)
                printDocument1.Print();
            //</show_dialog>
        }


        //<specify_the_material_to_be_printed>
        private void PrintDocument1_PrintPage(object sender, System.Drawing.Printing.PrintPageEventArgs e) =>
            e.Graphics.FillRectangle(Brushes.Red, new Rectangle(100, 100, 100, 100));
        //</specify_the_material_to_be_printed>

        //<message_box_indicating_document_has_finished_printing>
        private void PrintDocument1_EndPrint(object sender, System.Drawing.Printing.PrintEventArgs e) =>
            MessageBox.Show(printDocument1.DocumentName + " has finished printing.");
        //</message_box_indicating_document_has_finished_printing>

    }
}
